### DISCUSSION TITLE
Auditlog of CLI queries

### DISCUSSION THREAD

**Discussion by fshuman-a** [planetscale/discussion#562] [Q&A] [answered]
Hello,

Is there a way to get an audit log of all CLI queries (or all queries in general - not just CLI). 

Looking at query insights, it does not tell us who ran what query. I'm thinking of a report/table that gives us the following info:
- user
- ip address: not essential
- time of operation
- query

If this feature isn't currently in PlanetScale, do you have any ideas on how we can capture this? Moreover, is this something we can get in the future?

Thank you.

**Comment by orware** [comment, marked as answer]
Hello @fshuman-a,

Logging of all queries is not something we currently provide from the PlanetScale side of things as I believe it comes with both a hefty storage requirement and [additional privacy/data concerns](https://planetscale.com/docs/concepts/query-insights#complete-query-collection) to log that information even for short periods of time.

Our Insights area is limited to a 7 day timeframe on our paid plans, but it also purposefully focuses on just queries that are running too often, too long, returning too much data, producing errors, etc. rather than everything, making it a bit easier to focus on the queries that can be having an impact on general performance, but it isn't intended to query everything for audit logging type purposes.

Insights may not share specifically who ran a query from a logged in user perspective, but if only key/trusted users have access to the PlanetScale UI, and other users are provided credentials unique to them to access the database to run queries, you can filter Insights by the credential's username in a case like that.

Query logging can also be implemented at the application layer or with other tools, but it would still leave some gaps such as the queries ran directly from the CLI or in other ways.

I can't speak officially on the topic of whether or not this type of data could be more comprehensively logged in the future, but at the moment I would say that the likelihood probably isn't very high that logging all queries will become an option (it's always a possibility that could change though).

**Reply by fshuman-a** [reply]
Thank you @orware !
